### PR TITLE
Upgrade Code Analysis analyzers and fix results

### PR DESCRIPTION
**Pull request by davkean**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/project-system/pull/6374)

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editors/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -12,5 +12,4 @@ Shared My.Resources.GeneralOptionPageResources.General_FastUpToDateCheck_LogLeve
 Shared My.Resources.GeneralOptionPageResources.General_FastUpToDateCheck_Title() -> String
 Shared My.Resources.GeneralOptionPageResources.General_NestingIgnoreSolutionAndProjectProfiles() -> String
 Shared My.Resources.GeneralOptionPageResources.General_NestingSettings_Title() -> String
-Shared My.Resources.GeneralOptionPageResources.ResourceManager() -> System.Resources.ResourceManager
-XamlGeneratedNamespace.GeneratedInternalTypeHelper.New() -> Void
\ No newline at end of file
+Shared My.Resources.GeneralOptionPageResources.ResourceManager() -> System.Resources.ResourceManager
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/PublicAPI.Unshipped.txt]: Doesn't seem related to this commit. Is this intentional?

File: tests/.editorconfig [hunk 2]
```diff
@@ -13,6 +13,7 @@ csharp_style_var_elsewhere = true:none
 # where we want to marshal the continuations back to it.
 dotnet_diagnostic.CA2007.severity = none
 dotnet_diagnostic.CA1822.severity = none
+dotnet_diagnostic.CA2016.severity = none        # ForwardCancellationTokenToInvocations
```
Comment (drewnoakes) [inline tests/.editorconfig:16]: Suggestion?

Comment (davkean) [reply tests/.editorconfig:16]: It's all noise in the unit tests, hence turned it off just under it.

Comment (drewnoakes) [reply tests/.editorconfig:16]: Missed that this was in the tests file.
